src/hibernate-resume/hibernate-resume.c

```diff
@@ -109,16 +122,29 @@ static int run(int argc, char *argv[]) {
 
         umask(0022);
 
-        if (!in_initrd())
+        if (!in_initrd() && !arg_clear_efi)
+                return log_error_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE),
+                                       "Not running in initrd, refusing to initiate resume from hibernation.");
+        if (in_initrd() && arg_clear_efi) {
+                log_info("Running in initrd, --clear-efi has no effect, exiting.");
                 return 0;
+        }
 
         if (argc <= optind) {
                 r = setup_hibernate_info_and_warn();
                 if (r <= 0)
                         return r;
 
                 if (arg_info.efi)
-                        (void) clear_efi_hibernate_location_and_warn();
+                        r = clear_efi_hibernate_location_and_warn();
```

Comment:
Hm, on `--clear-efi` mode, it is not necessary to parse EFI variable and kernel command line, that is `setup_hibernate_info_and_warn()` above is not necessary. Moreover, I think we should clear the EFI var even if we fail to parse them.

I'd like to call `clear_efi_...()` and return earlier:
```c
static int action_clear_efi(void) {
        if (in_initrd()) {
                log_info(...);
                return 0;
        }

        r = clear_efi_...();
        if (r > 0)
                log_notice(...);
        return r;
}

static int run(...) {
        ...
        umask(0022);

        if (arg_clear_efi)
                return action_clear_efi();

        ...
}
```

Reply:
This is intentional. If people boot from USB drives/portable systems/... we really shouldn't erase this? If the system identifier mismatches the resume won't be attempted in the first place?

Reply:
I mean, originally I didn't want to add this bit of automation either. I imagined very few people running into this, yet that turned out not to be the case.

IOW, one can always manually delete the var in efivarfs if that's strictly necessary. Every failed resume attempt means the loss of some data, so I don't think we should relax this even further...

Reply:
Hm, yeah, may be. Let's wait for another reviewer.

Reply:
Hmm, but the parse of kernel cmdline should really go. Will update.